Add SDK for Vulnerability Assessment on Managed Instance

talhers opened the pull request:
<!-- DO NOT DELETE THIS TEMPLATE -->

## Description
<!--
Please add an informative description that covers that changes made by the pull request.

If you are regenerating your SDK based off of a new swagger spec, please add the link to the corresponding swagger spec pull request that has been merged in the azure-rest-api-specs repository
-->

Adding Vulnerability Assessment SDK on managed instance
PR for swagger - https://github.com/Azure/azure-rest-api-specs/pull/3353

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] Please add REST spec PR link to the SDK PR
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

talhers committed:
Add SDK for Vulnerability Assessment on Managed Instance

azuresdkci commented:
Can one of the admins verify this patch?

talhers commented:
@yaakoviyun @yoavfr @jaredmoo for reference

dsgouda commented:
@talhers Please link the related REST spec pr and resolve the conflicts

jaredmoo commented on src/SDKs/SqlManagement/Sql.Tests/ManagedDatabaseVulnerabilityAssessmentTestFixture.cs:
```diff
@@ -0,0 +1,54 @@
+                        AdministratorLoginPassword = SqlManagementTestUtilities.DefaultPassword,
+                        Sku = sku,
+                        SubnetId =
+                            "/subscriptions/a8c9a924-06c0-4bde-9788-e7b1370969e1/resourceGroups/RG_MIPlayground/providers/Microsoft.Network/virtualNetworks/VNET_MIPlayground/subnets/MISubnet",
```
This subnet id shouldn't be hardcoded. Tests must not require any precreated dependencies. Please create the subnet in test code.

talhers replied:
After consulting with the managed instance team I created the managed instance the same way that it is created here - https://github.com/Azure/azure-sdk-for-net/blob/psSdkJson6/src/SDKs/SqlManagement/Sql.Tests/ManagedInstanceCrudScenarioTests.cs

jaredmoo commented on src/SDKs/SqlManagement/Sql.Tests/ManagedDatabaseVulnerabilityAssessmentTestFixture.cs:
```diff
@@ -0,0 +1,54 @@
+
+namespace Sql.Tests
+{
+    public class ManagedDatabaseVulnerabilityAssessmentTestFixture : IDisposable
```
Since this is just creating a Managed Instance, consider renaming this to `ManagedInstanceTestFixture`

talhers replied:
I renamed it

talhers committed:
Merge branch 'psSdkJson6' into psSdkJson6

talhers committed:
Rename ManagedDatabaseVulnerabilityAssessmentTestFixture

Rename ManagedDatabaseVulnerabilityAssessmentTestFixture to ManagedInstanceTestFixture

talhers commented:
> @talhers Please link the related REST spec pr and resolve the conflicts

REST spec pr - https://github.com/Azure/azure-rest-api-specs/pull/3353

jaredmoo commented on src/SDKs/SqlManagement/Management.Sql/Microsoft.Azure.Management.Sql.csproj:
```diff
@@ -7,11 +7,12 @@
+- Support Vulnerability Assessment on Managed Instance
 - Added support for Managed Instance keys and encryption protectors.
 - Added support for DatabaseVulnerabilityAssessment.StorageAccountAccessKey, which causes DatabaseVulnerabilityAssessment.StorageContainerSasKey to no longer be a required property.
 - Added support for Managed Instance DnsZone, DnsZonePartner and Collation properties
```
Please wipe out the new features from last release.

jaredmoo commented on src/SDKs/SqlManagement/Sql.Tests/ManagedInstanceTestFixture.cs:
```diff
@@ -0,0 +1,54 @@
+                        Sku = sku,
+                        SubnetId =
+                            "/subscriptions/a8c9a924-06c0-4bde-9788-e7b1370969e1/resourceGroups/RG_MIPlayground/providers/Microsoft.Network/virtualNetworks/VNET_MIPlayground/subnets/MISubnet",
+                        Tags = new Dictionary<string, string>(),
```
I had a comment here that I don't want to be lost - please create the subnet as part of the test fixture. Otherwise the code cannot be executed by other than you :)

dsgouda commented:
@azuresdkci Retest this please

dsgouda commented:
@azuresdkci Test this please

dsgouda commented on src/SDKs/SqlManagement/Management.Sql/Generated/SdkInfo_SqlManagementClient.cs:
```diff
@@ -94,10 +97,10 @@ internal static partial class SdkInfo
       // BEGIN: Code Generation Metadata Section
       public static readonly String AutoRestVersion = "latest";
       public static readonly String AutoRestBootStrapperVersion = "autorest@2.0.4283";
-      public static readonly String AutoRestCmdExecuted = "cmd.exe /c autorest.cmd https://github.com/Azure/azure-rest-api-specs/blob/master/specification/sql/resource-manager/readme.md --csharp --version=latest --reflect-api-versions --csharp-sdks-folder=E:\\Github2\\azure-sdk-for-net\\src\\SDKs";
```
Please regenerate the code from spec merged in master

talhers replied:
done

talhers committed:
Test fixure creates everything (subnet id is not hardcoded)

talhers committed:
Wipe out new features from last release

talhers committed:
Generate code from spec merged in master

dsgouda reviewed (approved):
LGTM subject to CI builds

jaredmoo reviewed (approved):
Looks great!

dsgouda commented:
@azuresdkci Retest this please

dsgouda commented:
@talhers Please join the Azure org [here](aka.ms/azuregithub) in order to enable CI runs on this PR

talhers commented:
> @talhers Please join the Azure org [here](http://aka.ms/azuregithub) in order to enable CI runs on this PR

done
